librz/core/disasm.c

```diff
@@ -5292,17 +5366,19 @@ RZ_API int rz_core_print_disasm(RzPrint *p, RzCore *core, ut64 addr, ut8 *buf, i
 
 	// TODO: All those ds must be print flags
 	RzDisasmState *ds = ds_init(core);
-	ds->cbytes = cbytes;
+	ds->cbytes = options ? options->cbytes : 0;
 	ds->print = p;
-	ds->l = l;
+	ds->l = nlines;
 	ds->buf = buf;
 	ds->len = len;
 	ds->addr = addr;
 	ds->hint = NULL;
 	ds->buf_line_begin = 0;
-	ds->pdf = pdf;
+	ds->pdf = options ? options->function : NULL;
+	ds->pj = NULL;
+	ds->list = out_list;
```

Comment:
why are we getting the list from outside? Shouldn't it be initialized here? e.g. rz_list_newf(rz_disassemble_text_free) ?

Reply:
It's just a different way, I don't think there's anything wrong with me